### PR TITLE
Fix StatusUpdate regression from #665

### PR DESCRIPTION
**Pull request by dmvolod**

### CODE REVIEW

File: pkg/apis/infinispan/v1/types_util.go [hunk 1]
```diff
@@ -324,11 +324,13 @@ func (ispn *Infinispan) UpdateStatus(client client.Client, logger logr.Logger, s
 		if statusUpdate != nil {
 			statusUpdate()
 		}
-		err := client.Status().Update(context.TODO(), ispn)
+		ispn.Status.DeepCopyInto(&infinispan.Status)
```
Comment (ryanemerson) [inline pkg/apis/infinispan/v1/types_util.go:327]: Why are we explicitly calling `DeepCopyInto` here? Can't we just do `infinispan.status = ispn.status` and the value will be copied.

Comment (dmvolod) [reply pkg/apis/infinispan/v1/types_util.go:327]: `DeepCopy` do the same if `struc` don't have an elements we need special operation, like arrays, slices, pointers, etc.

Comment (dmvolod) [reply pkg/apis/infinispan/v1/types_util.go:327]: And I don't want to be care about all of this

**Comment by dmvolod** [comment]
@ryanemerson CI now green but output still has some sort of errors during update as part of StatusUpdate regression
I will plan to fix them as part of the #674

File: pkg/apis/infinispan/v1/types_util.go (hunk 1 above)
Comment (rigazilla) [reply pkg/apis/infinispan/v1/types_util.go:327]: ok by me

**Review by rigazilla — APPROVED** [review]

**Comment by ryanemerson** [comment]
Thanks @dmvolod

**Comment by rigazilla** [comment]
https://github.com/infinispan/infinispan-operator/blob/44890d0e7c97772c0bf31baa85f4722c31d6134e/pkg/apis/infinispan/v1/types_util.go#L323

Little detail
I haven't noticied before, but this line can be simpilfied

**Comment by dmvolod** [comment]
> https://github.com/infinispan/infinispan-operator/blob/44890d0e7c97772c0bf31baa85f4722c31d6134e/pkg/apis/infinispan/v1/types_util.go#L323
> 
> Little detail
> I haven't noticied before, but this line can be simpilfied

Probably yes, @rigazilla please suggest chnages

**Comment by rigazilla** [comment]
```
	if statusValidate == nil || (statusValidate != nil && statusValidate()) {
		if statusUpdate != nil {
			statusUpdate()
		}
```
I think that this is equivalent, due to how go evaluates boolean expr
```
	if statusValidate == nil  || statusValidate() {
		if statusUpdate != nil {
			statusUpdate()
		}
```

**Comment by dmvolod** [comment]
Cool, exactly, thanks, @rigazilla

**Pull request merged**